### PR TITLE
Entry point manifest now contains corresponding execution modes.

### PR DESCRIPTION
**Pull request by zedrian**
A new enum was added - `ExecutionMode` (filled with values from official Vulkan 1.2 SPIR-V specification). 
During SPIR-V reflection, declared execution modes are collected for all entry points and later stored in the corresponding `Manifest` objects, where they could later be accessed by the users to determine declared workgroup sizes, tessellation strategies, etc.

### CODE REVIEW

**Review by PENGUINLIONG — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -260,6 +260,7 @@ pub struct MemberVariableResolution<'a> {
 /// A set of information used to describe variable typing and routing.
 #[derive(Default, Clone)]
 pub struct Manifest {
+    execution_modes: Vec<ExecutionMode>,
```
[the remark this answers is not in the document]
Comment (zedrian) [reply src/lib.rs:263]: You are right. I'll move execution modes to `EntryPoint` then.

File: src/reflect.rs [hunk 2]
```diff
@@ -153,14 +153,211 @@ pub struct EntryPointDeclartion<'a> {
...
+    DepthUnchanged,
+    /// Indicates the work-group size in the x, y, and z dimensions.
+    ///
+    /// Only valid with the GLCompute or Kernel execution models.
+    LocalSize { x: u32, y: u32, z: u32 },
+    /// A hint to the compiler, which indicates the most likely to be used work-group size in the
+    /// x, y, and z dimensions.
+    ///
+    /// Only valid with the Kernel execution model.
+    LocalSizeHint { x: u32, y: u32, z: u32 },
```
[the remark this answers is not in the document]
Comment (zedrian) [reply src/reflect.rs:250]: Okay, I'll remove all Kernel-related variants from the enum for now. If SPIR-Q meets OpenCL support one day, such execution modes will just safely fall into the "unsupported execution mode" warning branch, so we will not forget to add them back.

[the remark this answers is not in the document]
Comment (PENGUINLIONG) [reply src/reflect.rs:250]: IIRC CL 2.x supports SPIR-V ingestion while few vendors have implemented full-feature OpenCL 2.x drivers. Nvidia and Imagination implemented only up to CL 1.2. And as per Khronos, CL 1.2 is the de facto standard of OpenCL; which means most of the applications still use OpenCL C for the programs. So I believe for a decent length of time it won't bother.

**Review by PENGUINLIONG — APPROVED** [review]

**Comment by PENGUINLIONG** [comment]
LGTM. Thanks for your contribution!

**Pull request merged**